test/parallel/test-quic-send-fd.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+const common = require('../common');
+if (!common.hasQuic)
+  common.skip('missing quic');
+
+const assert = require('assert');
+const quic = require('quic');
+const fs = require('fs');
+
+const fixtures = require('../common/fixtures');
+const key = fixtures.readKey('agent1-key.pem', 'binary');
+const cert = fixtures.readKey('agent1-cert.pem', 'binary');
+const ca = fixtures.readKey('ca1-cert.pem', 'binary');
+
+const server = quic.createSocket({ port: 0, validateAddress: true });
+
+server.listen({
+  key,
+  cert,
+  ca,
+  rejectUnauthorized: false,
+  maxCryptoBuffer: 4096,
+  alpn: 'meow'
+});
+
+server.on('session', common.mustCall((session) => {
+  session.on('secure', common.mustCall((servername, alpn, cipher) => {
+    const stream = session.openStream({ halfOpen: false });
+    stream.sendFD(fs.openSync(__filename, 'r'));
```

Comment:
Just been thinking about this for a while... wondering if the `sendFD()` shouldn't have a callback or is the `on('close')` event good enough?

Also, should have a variant that tests error conditions and ensures that the `on('error')` event is emitted properly.

Reply:
> Just been thinking about this for a while... wondering if the `sendFD()` shouldn't have a callback or is the `on('close')` event good enough?

I’d say `on('close')` is good enough.

> Also, should have a variant that tests error conditions and ensures that the `on('error')` event is emitted properly.

It’s in the PR TODO list :)